app/codec.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+package app;
+
+import "github.com/iov-one/weave/x/cash/codec.proto";
+import "github.com/iov-one/weave/x/sigs/codec.proto";
+import "gogoproto/gogo.proto";
+import "x/custom/codec.proto";
+
+// Tx contains the message
+message Tx {
+  // fee info, autogenerates GetFees()
+  cash.FeeInfo cash_fees = 20;
+  repeated sigs.StdSignature sigs_signatures = 21;
+  // ID of a multisig contract.
+  repeated bytes multisig = 4;
+  // msg is a sum type over all allowed messages on this chain.
+  oneof sum {
+    cash.SendMsg cash_send_msg = 51;
+    // space here to allow many more....
+
+    custom.CreateCustomStateIndexedMsg custom_create_state_indexed_msg = 100;
+    custom.CreateCustomStateMsg create_custom_state_msg = 101;
```

Comment:
If you add these, please update the Router and QueryHandler set in the app to include these

Reply:
I still need to implement handler and buckets. Lets keep this PR on until I finish and merge them